src/mapeo-project.js

```diff
@@ -749,6 +761,410 @@ export class MapeoProject extends TypedEmitter {
+        }
+      }
+
+      let latitude = lat
+      let longitude = lon
+      let altitude = null
+      const position = observation?.metadata?.position?.coords
+      if (position) {
+        latitude = position.latitude
+        longitude = position.longitude
+        if (position.altitude !== undefined) {
+          altitude = position.altitude
+        }
+      }
+
+      const coordinates = [longitude, latitude]
+      if (typeof altitude === 'number') {
+        coordinates.push(altitude)
+      }
+      const hasLatLon =
+        typeof longitude === 'number' && typeof latitude === 'number'
+      const geometry = hasLatLon
+        ? {
+            type: 'Point',
+            coordinates,
+          }
+        : null
+      const comma = first ? '' : ','
+      first = false
+      yield b4a.from(
+        `${comma}\n      ` +
+          JSON.stringify({
+            type: 'Feature',
+            properties: observation,
+            geometry,
+          })
+      )
+    }
+  }
+
+  /**
+   * @param {Iterable<Track>} tracks
+   * @param {Object} options
+   * @param {Set<string>} [options.seenObservations=new Set()]
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @param {string} [options.lang]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportTracks(
+    tracks,
+    { lang, seenObservations = new Set(), seenAttachments = new Map() } = {}
+  ) {
+    let first = true
+    for (const track of tracks) {
+      const { observationRefs } = track
+
+      const observations = await Promise.all(
+        observationRefs.map(({ docId }) =>
+          this.#dataTypes.observation.getByDocId(docId, { lang })
+        )
+      )
+
+      const coordinates = track.locations.map(
+        ({ coords: { longitude, latitude, altitude } }) => [
+          longitude,
+          latitude,
+          altitude,
+        ]
+      )
+      const comma = first ? '' : ','
+      first = false
+      yield b4a.from(
+        `${comma}\n      ` +
+          JSON.stringify({
+            type: 'Feature',
+            properties: track,
+            geometry: {
+              type: 'LineString',
+              coordinates,
+            },
+          }) +
+          '\n'
+      )
+
+      let firstObs = true
+      for await (const chunk of this.#exportObservations(observations, {
+        seenObservations,
+        seenAttachments,
+      })) {
+        if (firstObs) {
+          yield b4a.from(',')
+          firstObs = false
+        }
+        yield chunk
+      }
+    }
+  }
+
+  /**
+   * @param {Object} [options={}]
+   * @param {boolean} [options.observations=true] Whether observations should be exported
+   * @param {boolean} [options.tracks=true] Whether all tracks and their observations should be exported
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @param {string} [options.lang]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportGeoJSONIterator({
```

Comment:
It would be worth consulting users once this feature is released whether it is complicated mixing tracks and observations in the same geojson file. From my knowledge when importing into GIS software, mixing geometry types in a file can sometimes make data processing more complicated.

Reply:
From the UI you choose only one or the other right now.